### PR TITLE
Add route for errors with secrets and PII

### PR DESCRIPTION
**Pull request by heaths**
Relates to Azure/azure-sdk#3959

### CODE REVIEW

File: src/test-routes/insecure-error.ts
Comment (christothes) [inline src/test-routes/insecure-error.ts:5]: /errorWithSecret ?

**Comment by heaths** [comment]
Per @lmazuel added swagger and updated instructions for future devs to do likewise.

File: .editorconfig
Comment (timotheeguerin) [inline .editorconfig:12]: Probably just move that to the `*` section as its enabled for all files in gittattributes

**Review by timotheeguerin — APPROVED** [review]
Will need to bump the patch version in packge.json to get it published.

File: .editorconfig
Comment (heaths) [reply .editorconfig:12]: From line 5:

> # (Please don't specify an indent_size here; that has too many unintended consequences.)

I don't disagree in principle, but don't want this PR to be the thing that could break a bunch.

File: swagger/error-with-secrets.json
Comment (lmazuel) [inline swagger/error-with-secrets.json:17]: This swagger means all calls are failure, autorest refuses this and won't generate in some language since that's an impossible scenario. At least add a 204 with no body (that's weird for a GET, but least describe a success scenario)

**Comment by heaths** [comment]
> Will need to bump the patch version in packge.json to get it published.

I did but the rebase dropped it since you had already, apparently. Do we need to do it yet again i.e., is the push automatic from a release pipeline?

**Review by lmazuel — APPROVED** [review]

**Comment by heaths** [comment]
I want to hold off on actually merging this till I heard back from some of the architects involved in the related archboard issue.

**Comment by timotheeguerin** [comment]
> @timotheeguerin
> 
> > Will need to bump the patch version in packge.json to get it published.
> 
> I did but the rebase dropped it since you had already, apparently. Do we need to do it yet again i.e., is the push automatic from a release pipeline?

Yeah it is a little of a race condition here with the bumping. Before merging you should check that you are up are bumping relative to what is in main otherwise it won't get published.

File: .editorconfig
Comment (timotheeguerin) [reply .editorconfig:13]: Ended-up doing that one in another PR as it was quite needed after the git change. I think you can remove that line now

File: src/test-routes/error-with-secrets.ts
Comment (timotheeguerin) [inline src/test-routes/error-with-secrets.ts:34]: is this supposed to be a parameter? I see it is also like that in the swagger file.
express.js will assume you have  a path param called `create` here.

Comment (heaths) [reply src/test-routes/error-with-secrets.ts:34]: It's an action, following Azure guidelines.

Comment (heaths) [reply src/test-routes/error-with-secrets.ts:34]: `/secrets/:create` is valid: https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#performing-an-action, though, @JeffreyRichter @mikekistler I see that document wasn't updated to say that `/:` was allowed now that the Postman issue I raised has been resolved.

curl returns 303 if I try http://localhost:3000/secrets/notcreate so express seems to support it just fine as well.

**Comment by heaths** [comment]
@bterlson @johanste you good with these endpoints? Keeping up with the constant package.json semver changes is becoming a hassle, so I'd like to get this in ASAP.

File: src/test-routes/error-with-secrets.ts
Comment (heaths) [reply src/test-routes/error-with-secrets.ts:34]: Sorry, I didn't set the write method on the attempt. Using POST "notcreate" does "work", but escaping with `[:]` (per online docs) works correctly.

**Comment by heaths** [comment]
Because package.json keeps getting updated, I'm merging this now since it's signed off. If someone wants to leave comments I'll take a look and can submit a new PR.

File: src/test-routes/error-with-secrets.ts
Comment (mikekistler) [reply src/test-routes/error-with-secrets.ts]: The Azure Guidelines are to use a ":" with no preceding "/". We discussed allowing "/:" and decided against it since we want to promote a single pattern wherever possible for consistency.

Comment (heaths) [reply src/test-routes/error-with-secrets.ts]: Thanks. Based on https://github.com/microsoft/api-guidelines/issues/265 I see I misremembered. Do you feel this should be changed in this test service to reflect that, or not a big deal? Happy to submit a PR if it should be changed.

Comment (mikekistler) [reply src/test-routes/error-with-secrets.ts]: Not a big deal for this to be in the test server -- I would not go to the trouble of changing it now.

**Pull request merged**
